packages/core/src/contrib/Constraints.ts

```diff
@@ -229,6 +230,51 @@ export const objDict = {
   nearScalar: (c: any, goal: any) => {
     return squared(sub(constOfIf(c), constOfIf(goal)));
   },
+  /**
+   * Repel the angle between the p1-p0 and p1-p2 away from 0 and 180 degrees.
+   * NOTE: angles more than `range` degrees from 0 or 180 deg are considered satisfied.
+   */
+  nonDegenerateAngle: (
+    [, p0]: [string, any],
+    [, p1]: [string, any],
+    [, p2]: [string, any],
+    strength = 20,
+    range = 10
+  ) => {
+    if (every([p0, p1, p2].map((props) => props["center"]))) {
+      const c0 = fns.center(p0);
+      const c1 = fns.center(p1);
+      const c2 = fns.center(p2);
+
+      const l1 = ops.vsub(c0, c1);
+      const l2 = ops.vsub(c2, c1);
+      const cosine = absVal(ops.vdot(ops.vnormalize(l1), ops.vnormalize(l2)));
+      // angles that are more than `range` deg from 0 or 180 do not need to be pushed
+      return ifCond(
+        lt(cosine, varOf(range * (Math.PI / 180))),
+        varOf(0),
+        mul(constOfIf(strength), cosine)
+      );
+    } else {
+      throw new Error(
+        "nonDegenerateAngle: all input shapes need to have centers"
+      );
+    }
+  },
+  /**
+   * try to make distance between a point and a segment `s1` = padding.
+   */
+  pointLineDist: ([, s1]: [string, any], point: VarAD[], padding: VarAD) => {
```

Comment:
Apologies for the back and forth here, but if you could switch the arguments to this objective so the first argument is the point and the second argument is the line that would be awesome. Also please test if the line `isLineLike` and throw an error if it's not. (check the other constraints and objectives to see how this is done)